File: MVVMSample/app/src/main/java/com/example/mvvmsample/data/networking/APIService.java
package com.example.mvvmsample.data.networking;

import com.example.mvvmsample.data.models.Todo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.Url;

public interface APIService {

    @GET("todos/")
    Call<List<Todo>> getTodos();


    @GET("todos/{id}/")
    Call<Todo> getTodoById(@Path("id") int todoId);


    @Headers({"Cache-Control: max-age=640000", "User-Agent: My-App-Name"})
    @GET("todos/")
    Call<List<Todo>> getTodosByUserId(@Query("userId") int userId,
                                      @Query("completed") boolean completed);


    @FormUrlEncoded
    @POST("todos/")
    Call<Todo> addTodo(@Field("userId") int userId,
                       @Field("title") String title,
                       @Field("completed") boolean completed);



    @GET
    Call<List<Todo>> getTodosFromUrl(@Url String url);

}

File: RetrofitApplication/app/src/main/java/com/example/retrofitapplication/ApiService.kt
package com.example.retrofitapplication

import retrofit2.Call
import retrofit2.http.*

interface ApiService {

    @GET("todos/")
    fun getTodos(): Call<List<Todo>>

    @GET("todos/{id}/")
    fun getTodoById(@Path("id") todoId: Int): Call<Todo>

    @Headers("Cache-Control: max-age=640000", "User-Agent: My-App-Name")
    @GET("todos/")
    fun getTodosByUserId(
        @Query("userId") userId: Int,
        @Query("completed") completed: Boolean
    ): Call<List<Todo>>


    @FormUrlEncoded
    @POST("todos/")
    fun addTodo(
        @Field("userId") userId: Int,
        @Field("title") title: String,
        @Field("completed") completed: Boolean
    ): Call<Todo>

    @GET
    fun getTodosWithUrl(@Url url: String): Call<List<Todo>>
}